### DISCUSSION TITLE
Different header height on first page

### DISCUSSION THREAD

**Discussion by pkacp** [prawnpdf/prawn#1331] [Q&A]
Hello,
I'm trying to build a header that starts on second page of a document.
How to do it in the way so the texts starts at the very top of first page and on the second and following pages starts after the header?
I'm using the `dynamic: true` option for `repeat` so it does't leave blank space on first page.
Wrapping text into the `bounding_box` seems as no go solution as its going to start at the same height on each page. Maybe there is a conditional `bounding_box` or `span`?

```ruby
require "prawn"

Prawn::Document.generate("sample.pdf") do
  # header
  repeat(->(page_number) { (page_number > 1) }, dynamic: true) do
    bounding_box [bounds.left, bounds.top], :width  => bounds.width do
      text "Some Header ", align: :center, size: 33
      text "additional text for the header", align: :center
      stroke_horizontal_rule
    end
  end

  text "Lorem ipsum"*200
  text "Lorem ipsum"*600
  text "Lorem ipsum"*200
end
```
<img width="1110" alt="Screenshot 2024-01-26 at 15 08 06" src="https://github.com/prawnpdf/prawn/assets/40572695/6a4a8549-d828-4f8e-8e73-4677d68cb2eb">

**Comment by pointlessone** [comment]
I'd advise against using a repeater. For your specific case `on_page_create` hook would work great.

**Reply by pkacp** [reply]
Any advice what to use to move the text below the header? because currently it works exactly the same as with `repeat`; the text is not flowing after the header. The `move_down` and `move_cursor_to` have no effect 
```ruby
  # header
  on_page_create do
    bounding_box [bounds.left, bounds.top], :width => bounds.width do
      text "Some Header ", align: :center, size: 33
      text "additional text for the header", align: :center
      stroke_horizontal_rule
    end
    move_down(200)
    move_cursor_to(500)
  end
```

<img width="1124" alt="image" src="https://github.com/prawnpdf/prawn/assets/40572695/c572b3b2-214a-474e-bb55-9dbf7cc3f250">

**Reply by pkacp** [reply]
@pointlessone sorry to bother you, but can you give me any tips for solving that one?

**Reply by pointlessone** [reply]
The callback is ran in a `float` so you'd have to save cursor in a variable outside the block. For example, in an instance variable. Or you can set an appropriate bounds on the new page in the callback based on the header height.
